### PR TITLE
Fixing builder set date

### PR DESCRIPTION
**Pull request by jivewise**
Fixing the bug described here: https://www.pivotaltracker.com/n/projects/1156756/stories/103133164

The bug was due to calling setDueAt and setOpensAt with a date object instead of a moment.  It should be able to handle both now.

Also adding in some specs.

### CODE REVIEW

**Comment by philschatz** [comment]
When I select a date clicking the Calendar (or the little popup calendar) the due date is set to 1 day before the date I selected.

**Comment by jivewise** [comment]
That bug should be fixed, also added specs to make sure it doesn't happen again.

**Comment by philschatz** [comment]
When I select tomorrow (9/17) from the Calendar the due date is still (9/16)  (maybe timezone related?):

image


The little popup calendar however does seem to work properly now ; )

**Comment by jivewise** [comment]
That calendar error was due to us setting the default dates before we set the correct timezone in the componentWillMount function.  That should be good to go now.

File: src/components/task-plan/builder.cjsx [hunk 1]
```diff
@@ -69,6 +70,7 @@ module.exports = React.createClass
     opensAt
 
   getQueriedDueAt: ->
+    window.moment = moment
```
Comment (philschatz) [inline src/components/task-plan/builder.cjsx]: Is this line needed?

**Comment by philschatz** [comment]
can you add a test to check that the dates are set correctly when mounted?

**Comment by jivewise** [comment]
Yep, adding a test now.

**Pull request merged**